src/controller_examples/zookeeper_controller/proof/liveness.rs

```diff
@@ -124,17 +124,196 @@ proof fn lemma_true_leads_to_always_the_object_in_schedule_has_spec_as(spec: Tem
+                    let post = |s: ClusterState| {
+                        &&& !s.reconcile_state_contains(zk.object_ref())
+                        &&& s.reconcile_scheduled_for(zk.object_ref())
+                    };
+                    let input = zk.object_ref();
+
+                    lemma_always_desired_state_exists(stable_spec_with_assumption, zk);
+                    temp_pred_equality::<ClusterState>(
+                        lift_state(desired_state_exists(zk)), lift_state(schedule_controller_reconcile().pre(input))
+                    );
+                    spec_implies_pre(stable_spec_with_assumption, lift_state(pre), lift_state(schedule_controller_reconcile().pre(input)));
+                    use_tla_forall::<ClusterState, ObjectRef>(stable_spec_with_assumption, |key| schedule_controller_reconcile().weak_fairness(key), input);
+                    schedule_controller_reconcile().wf1(input, stable_spec_with_assumption, stronger_next, pre, post);
+                    leads_to_trans_temp(stable_spec_with_assumption, lift_state(pre), lift_state(post), lift_state(safety::the_object_in_reconcile_has_spec_as(zk)));
+                }
+            );
+
+            or_leads_to_combine_temp(
+                stable_spec_with_assumption,
+                lift_state(|s: ClusterState| {
+                    &&& !s.reconcile_state_contains(zk.object_ref())
+                    &&& s.reconcile_scheduled_for(zk.object_ref())
+                }),
+                lift_state(|s: ClusterState| {
+                    &&& !s.reconcile_state_contains(zk.object_ref())
+                    &&& !s.reconcile_scheduled_for(zk.object_ref())
+                }),
+                lift_state(safety::the_object_in_reconcile_has_spec_as(zk))
+            );
+
+            temp_pred_equality(
+                lift_state(|s: ClusterState| {
+                    &&& !s.reconcile_state_contains(zk.object_ref())
+                    &&& s.reconcile_scheduled_for(zk.object_ref())
+                }).or(lift_state(|s: ClusterState| {
+                    &&& !s.reconcile_state_contains(zk.object_ref())
+                    &&& !s.reconcile_scheduled_for(zk.object_ref())
+                })),
+                lift_state(|s: ClusterState| !s.reconcile_state_contains(zk.object_ref()))
+            );
+
+            leads_to_trans_temp(
+                stable_spec_with_assumption,
+                true_pred(),
+                lift_state(|s: ClusterState| !s.reconcile_state_contains(zk.object_ref())),
+                lift_state(safety::the_object_in_reconcile_has_spec_as(zk))
+            );
+            leads_to_stable_temp(stable_spec_with_assumption, lift_action(stronger_next), true_pred(), lift_state(safety::the_object_in_reconcile_has_spec_as(zk)));
+        }
+    );
+
+    // By unpacking the conditions we will have: stable_spec |= []the_object_in_schedule_has_spec_as(zk) ~> []the_object_in_reconcile_has_spec_as(zk)
+    assume(valid(stable(stable_spec)));
```

Comment:
`assume`?

Reply:
Yes, I leave this proof as assumed for now. I find it quite cumbersome to prove stability so I plan to prove all the stability in the liveness proof later. You will find other such `assume()` in the liveness proof.